### PR TITLE
content/blog: Add second GSoC blog post about Multiboot2

### PR DESCRIPTION
**Pull request by mariapana**
Added the second GSoC blog post about Multiboot2 showcasing the progress made during the second set of 3 weeks.

Signed-off-by: Maria Pana <maria.pana4@gmail.com>

### CODE REVIEW

File: content/blog/2024-07-10-gsoc-multiboot2.mdx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+Additionaly, in the context of this project, it may be valuable to note that simply adding the Multiboot2 header to the ELF file is not sufficient.
+On top of that, both `multiboot.py` and `multiboot2.py` scripts add an extra ELF header, "sandwiching" the Multiboot2 header between it and the original ELF.
+This ensures that GRUB identifies the overall file as an ELF and avoids any potential complications along the way.
+
+### Interaction between Bootloader and Kernel
+
+The bootloader locates the Multiboot2 header within the ELF file.
+After verifying the magic number and architecture compatibility, it parses the header and tags to gather essential system information.
+It then loads the kernel into memory, sets up the environment, and transfers control to the kernel's entry point.
```
Comment (cristian-vijelie) [inline content/blog/2024-07-10-gsoc-multiboot2.mdx:90]: I would say here that the bootloader, after finding the information requested by the tags in the header, places that information in the kernel memory, then passes the control, for clarity

**Comment by StefanJum** [comment]
@mariapana please squash the commits. Otherwise looks fine.
@cristian-vijelie @mogasergiu after you take a look at it content-wise, we can merge.

File: content/blog/2024-07-10-gsoc-multiboot2.mdx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+- The target architecture  (e.g. `i386`, `x86_64`)
+- The header length, including all tags
+- The checksum of the header, used for error detection
+
+I talked a little about the tags in my previous blog post, but to recap, they provide detailed information about the system's memory layout, boot command line, and other essential parameters.
+Commonly, each tag has a type, size and payload.
+There is a multitude of tags to choose from, depending on the specific requirements of the system, adding to the flexibility of Multiboot2.
+
+Additionaly, in the context of this project, it may be valuable to note that simply adding the Multiboot2 header to the ELF file is not sufficient.
+On top of that, both `multiboot.py` and `multiboot2.py` scripts add an extra ELF header, "sandwiching" the Multiboot2 header between it and the original ELF.
```
Comment (mogasergiu) [inline content/blog/2024-07-10-gsoc-multiboot2.mdx:83]: > As discussed, you will probably still have to duplicate (and update) the 64-bit ELF headers.

Ah you mention it here as well, yeah, ok so then I would update what is said above as well I guess?
(also "sandwiching" :eye: :sandwich: :+1: )

**Review by mogasergiu — CHANGES_REQUESTED** [review]
Looks good! :+1: A couple of minor comments.

**Review by mogasergiu — APPROVED** [review]
LGTM.

Reviewed- by: Sergiu Moga <sergiu@unikraft.io>

**Review by cristian-vijelie — APPROVED** [review]

**Review by StefanJum — APPROVED** [review]
Reviewed-by: Stefan Jumarea <stefanjumarea02@gmail.com>
Approved-by: Stefan Jumarea <stefanjumarea02@gmail.com>

**Pull request merged**